### DISCUSSION TITLE
Usability when tracking down failures with moderately long expressions

### DISCUSSION THREAD

**Discussion by claremacrae** [jglev/obsidian-apply-patterns-plugin#28] [Q&A]
Many thanks for this plugin.

### Background

I've written tests for this Python re:

`\[(GitHub - ){0,1}([A-Za-z-\d)]+)/([a-z\d\-_.]+)(: [^\]]+){0,1}]\(https://github.com/([A-Za-z-\d)]+)/([a-z\d\-_.]+)/?\)`

And now I'm trying to port it to JS , to work with Apply Patterns.

### What's happening

I've got an expression which works in https://regexr.com

`\[(GitHub - ){0,1}([A-Za-z-\d)]+)\/([a-z\d\-_.]+)(: [^\]]+){0,1}]\(https:\/\/github.com\/([A-Za-z-\d)]+)\/([a-z\d\-_.]+)\/?\)`

![image](https://user-images.githubusercontent.com/4840096/137806078-ce46e223-609f-4f5c-b1f2-ace20cb02b90.png)

But when I test it in Apply Patterns I get this:

![image](https://user-images.githubusercontent.com/4840096/137805306-c0366f69-d56c-42b4-8ea7-c54579e3fb5d.png)

(I've gone through lots of iterations, so seen lots of variations of that kind of message)

And here is what I see in the settings:

![image](https://user-images.githubusercontent.com/4840096/137805575-4da3727c-c87c-485d-b726-beae4458378b.png)

I have found explanations of Lone Quantifier brackets - but I'm struggling to understand why an expression that works in regexr might not work on Apply Patterns...

### Thoughts

With the combination of only being able to see less than half of my expression in the pattern - and the very long error message only flashing up for a few seconds - I'm still going after some hours to get this working...

~~I'd really appreciate any advice on fixing that query~~ - but more generally, I wonder whether it might be possible for the layout to be adjust to make more use of the screen width, when it's available?

**Comment by claremacrae** [comment]
I got it working in the end, by pruning back the pattern from the end, to see at work point it starting compiling... and then working forward.

regexr does not require raw `]` to be escaped - Apply Patterns requires them to be `\]`.

I would still appreciate any thoughts on the layout - and whether perhaps the error message could be displayed inside the settings pane?

**Comment by jglev** [comment]
This response is too late to be of use, I'm sure; I'm sorry for not responding to this until now. With that said, I'm coming back and updating the plugin for the first time in quite a while, and so thought to post an update: v2.2.0 (#69) switches to displaying compiled regexes in a way that allows selecting and copying the text.

**Reply by claremacrae** [reply]
Thank you!
